### ISSUE TITLE
[Feature Request] Dynamic list capability support in tool contract

### ISSUE THREAD

**Issue by lisagreenview** [microsoft/promptflow#383]
I noticed that the LLM has the capability to dynamically list the deployment names based on the AOAI connection provided. It would be advantageous to incorporate this capability into the tool contract, allowing other tools, including custom tools, to benefit from it. For instance, the "Vector index lookup" tool could have the "path" input dynamically list the available ML Indexes in the workspace.

In summary, any tool should be able to define the listing experience and the corresponding request to enable the dynamic listing feature.

![image](https://github.com/microsoft/promptflow/assets/10419778/25e59071-f97a-4686-8943-59bc41a1721c)

**Comment by DaweiCai** [comment]
@chjinche is working on it.
